### PR TITLE
feat: add create/delete buckets from tree view

### LINKED ISSUE #279: Delete a bucket via the tree view
DOD:

  - Right click on a bucket in the tree view to delete it

### PR DESCRIPTION
**Pull request by rockstar**
This patch adds create/delete buckets functionality to the tree view.

Closes #279

### CODE REVIEW

**Comment by rockstar** [comment]
Please only look at the last commit from this patch, as it is based on the work in #296. I will not merge this patch until #296 has merged.

**Review by skartikey — APPROVED** [review]

**Pull request merged**
